lms/djangoapps/grades/rest_api/v1/gradebook_views.py

```diff
@@ -1044,44 +1047,49 @@ def get(self, request, subsection_id):
                 developer_message='Invalid UserID',
                 error_code='invalid_user_id'
             )
-
+        psg_exists = True
+        override = None
+        history = []
+        original_grade = None
         try:
             original_grade = PersistentSubsectionGrade.read_grade(user_id, usage_key)
-        except PersistentSubsectionGrade.DoesNotExist:
-            results = SubsectionGradeResponseSerializer({
-                'original_grade': None,
-                'override': None,
-                'history': [],
-                'subsection_id': usage_key,
-                'user_id': user_id,
-                'course_id': None,
-            })
-
-            return Response(results.data)
+            grade_data = {
+                'earned_all': original_grade.earned_all,
+                'possible_all': original_grade.possible_all,
+                'earned_graded': original_grade.earned_graded,
+                'possible_graded': original_grade.possible_graded,
+            }
 
-        limit_history_request_value = request.GET.get('history_record_limit')
-        if limit_history_request_value is not None:
+        except PersistentSubsectionGrade.DoesNotExist:
+            psg_exists = False
+            student = get_user_model().objects.get(id=user_id)
+            course_structure = get_course_blocks(student, usage_key)
+            subsection_grade_factory = SubsectionGradeFactory(student, course_structure=course_structure)
+            grade = subsection_grade_factory.create(course_structure[usage_key], read_only=True, force_calculate=True)
+            grade_data = {
+                'earned_all': grade.all_total.earned,
+                'possible_all': grade.all_total.possible,
+                'earned_graded': grade.graded_total.earned,
+                'possible_graded': grade.graded_total.possible,
+            }
+        history_record_limit = request.GET.get('history_record_limit')
+        if history_record_limit is not None:
             try:
-                history_record_limit = int(limit_history_request_value)
+                history_record_limit = int(history_record_limit)
             except ValueError:
                 history_record_limit = 0
 
-        try:
-            override = original_grade.override
-            if limit_history_request_value is not None:
-                history = reversed(list(override.history.all().order_by('-history_date')[:history_record_limit]))
-            else:
-                history = override.history.all().order_by('history_date')
-        except PersistentSubsectionGradeOverride.DoesNotExist:
-            override = None
-            history = []
+        if psg_exists:
+            if original_grade is not None and hasattr(original_grade, 'override'):
+                override = original_grade.override
+                history = list(original_grade.override.history.all().order_by('history_date')[:history_record_limit])
```

Comment:
You could eliminate the `psg_exists` variable and eliminate the initialization of `original_grade = None` if you moved these lines directly under line 1055, i.e.
```python
try:
    original_grade = PersistentSubsectionGrade.read_grade(user_id, usage_key)
    if hasattr(original_grade, 'override'):
        override = original_grade.override
        history = list(original_grade.override.history.all().order_by('history_date')[:history_record_limit])
```